packages/actor-scraper/camoufox-scraper/src/internals/crawler_setup.ts

```diff
@@ -182,14 +180,22 @@ export class CrawlerSetup implements CrawlerSetupOptions {
             return req;
         });
 
-        this.requestList = await RequestList.open(
-            'PLAYWRIGHT_SCRAPER',
-            startUrls,
-        );
-
         // RequestQueue
         this.requestQueue = await RequestQueueV2.open(this.requestQueueName);
 
+        const requests: Request[] = [];
+        for await (const request of await RequestList.open(null, startUrls)) {
+            if (
+                this.input.maxResultsPerCrawl > 0 &&
+                requests.length >= 1.5 * this.input.maxResultsPerCrawl
```

Comment:
when i mentioned the overenqueuing for 150% of the limit, i was mainly concerned about the requests added via `enqueueLinks`, i don't think its common that people provide more start URLs than the limit itself, the issue is what we enqueue on those pages. i guess we'd have to handle this on crawlee level.

Reply:
> i don't think its common that people provide more start URLs than the limit itself

Bold claim. I can imagine for example

1. someone setting a limit to get a quick sample of the results when developing their `pageFunction`
2. parsing URLs from an external source that's out of control of the user - it would make sense to add a limit in that case to prevent mayhem

The 150% is there because the crawl may fail for some URLs.

Reply:
i am not saying we shouldnt handle this here, i am saying the important part is elsewhere

Reply:
+1 for processing this in Crawlee, see https://github.com/apify/crawlee/issues/2728. 

Also, `this.input.maxResultsPerCrawl` is slightly confusing naming here - even if `pageFunction` returns `null`, it generates an "empty" dataset record (see https://github.com/apify/apify-sdk-js/issues/353).

IIRC `maxResultsPerCrawl` here = `maxRequestsPerCrawl` in Crawlee.